### ISSUE TITLE
Possible deadlock when using multiple instances of RtAudio / CoreAudio

### ISSUE THREAD

**Issue by tucano42** [thestk/rtaudio#242]
I can reliably hit a deadlock on RtAudio running on macOS 10.15:
1. open first instance of RtAudio for playback
2. create second instance of RtAudio and call RtAudio::getDeviceInfo

This will often deadlock in the call to AudioObjectGetPropertyDataSize

Here is a bug report from another project about a similar issue: https://forum.juce.com/t/coreaudio-possible-deadlock-when-moving-to-mavericks/11753

A google search for "AudioObjectGetPropertyDataSize deadlock" finds several other similar reports, and some solutions (which usually involve maintaining a copy of the device info instead of doing a live query all the time).

**Comment by tucano42** [comment]
Reading some of the other issues here, I get the impression that RtAudio does not yet fully support handling device connection and removal. This means caching the device info is as safe as a live query with the current code model. 
With that,  I will try to change RtAudioCore so there is a static variable std::vector<RtAudio::DeviceInfo>, and that any call to getDeviceInfo() will check if the number of devices reported by getDeviceCount() matches the size of the vector, and if the entry in the vector has already been populated, and if yes, return that one.

I have other code on my side that can detected device connection and removal, and that will trigger a rebuilt of the device info.

**Comment by tucano42** [comment]
Also, I just saw that the documentation states that "It is not possible to have multiple instances of RtAudio accessing the same CoreAudio device.".
Apart from the deadlock problem I described above, having multiple instances of RtAudio accessing the same CoreAudio device seems to work just fine. My use case for this is to play multiple audio streams with potentially different parameters (samplerate, channel number), or to play something like a metronome signal in addition to a recording.
